homeassistant/components/electric_kiwi/select.py

```diff
@@ -0,0 +1,74 @@
+"""Support for Electric Kiwi hour of free power."""
+from __future__ import annotations
+
+import logging
+
+from homeassistant.components.select import SelectEntity, SelectEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import ATTRIBUTION, DOMAIN
+from .coordinator import ElectricKiwiHOPDataCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+ATTR_EK_HOP_SELECT = "hop_select"
+
+HOP_SELECT = SelectEntityDescription(
+    entity_category=EntityCategory.CONFIG,
+    key=ATTR_EK_HOP_SELECT,
+    translation_key="hopselector",
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Electric Kiwi select setup."""
+    hop_coordinator: ElectricKiwiHOPDataCoordinator = hass.data[DOMAIN][entry.entry_id]
+
+    _LOGGER.debug("Setting up HOP entity")
+    entities = [ElectricKiwiSelectHOPEntity(hop_coordinator, HOP_SELECT)]
+    async_add_entities(entities)
+
+
+class ElectricKiwiSelectHOPEntity(
+    CoordinatorEntity[ElectricKiwiHOPDataCoordinator], SelectEntity
+):
+    """Entity object for seeing and setting the hour of free power."""
+
+    entity_description: SelectEntityDescription
+    _attr_has_entity_name = True
+    _attr_attribution = ATTRIBUTION
+    values_dict: dict[str, int]
+
+    def __init__(
+        self,
+        hop_coordinator: ElectricKiwiHOPDataCoordinator,
+        description: SelectEntityDescription,
+    ) -> None:
+        """Initialise the HOP selection entity."""
+        super().__init__(hop_coordinator)
+        self._attr_unique_id = f"{self.coordinator._ek_api.customer_number}_{self.coordinator._ek_api.connection_id}_{description.key}"
+        self.entity_description = description
+        self._state = None
+        self.values_dict = self.coordinator.get_hop_options()
+        self._attr_options = list(self.values_dict.keys())
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Update attributes when the coordinator updates."""
+        super()._handle_coordinator_update()
```

Comment:
Isn't this the same as actually removing it?

Reply:
I'll check and if so I'll remove it

Reply:
removed as its part of the coordinator entity